src/mg/PAMI/Client/Impl/ClientImpl.php

```diff
@@ -443,11 +439,14 @@ public function close()
 	 */
 	public function __construct(array $options)
 	{
-        if (isset($options['log4php.properties'])) {
-            \Logger::configure($options['log4php.properties']);
-        }
-        $this->_logger = \Logger::getLogger('Pami.ClientImpl');
-	    $this->_host = $options['host'];
+		$this->setLogger(
+			new Log4phpAdapter(
```

Comment:
We can make it an "optional option", and not break older implementations, allowing users to use other loggers, keeping the old behavior available when the new option is not available (just thinking out loud).

The hardcoding was done because there was not a standard interface for loggers, and log4php offered some cool features that loggers didn't offer at the time, so hardcoding the logger made more sense than writing an arbitrary logger interface not used by others in the community.

Since in this case we are dealing with a PHP standard, it makes more sense to use that as an interface and have the user provide the implementation, without making pami hardcoding a specific implementator.

From my point of view, writing an adapter and hardcoding it here seems to accomplish the same as leaving things as they are, since users can't change implementations.

I know, the user can later on change the implementation via setter injection, but it will force the user to momentarily use log4php (when instantiating the client) until the logger is set.

Just pointing out these details.